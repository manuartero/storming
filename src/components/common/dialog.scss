.dialog {
  border-radius: 10px;
  box-shadow: 0px 10px 50px hsla(0, 0%, 0%, 0.33);

  &--regular {
    width: 60%;
    height: 80%;
  }

  &--small {
    width: 50%;
    height: 60%;
  }
}
