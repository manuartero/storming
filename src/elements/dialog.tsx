import c from "classnames";
import { useId } from "react";

import styles from "./dialog.module.css";

type Props = {
  title?: string;
  size?: "regular" | "small";
  onClose: () => void;
} & React.ComponentProps<"div">;

export function Dialog({
  title,
  size = "regular",
  children,
  onClose,
  ...rest
}: Props) {
  const titleId = title ? `dialog-title-${useId()}` : undefined;

  return (
    <div
      role="dialog"
      aria-modal="true"
      aria-labelledby={title ? titleId : undefined}
      className={c(styles.dialogScreen)}
      tabIndex={-1}
      {...rest}
    >
      <div
        className={c(
          styles.dialog,
          size === "small" ? styles.small : styles.regular
        )}
      >
        <button
          className={styles.exitButton}
          aria-label="close dialog"
          onClick={onClose}
        >
          X
        </button>
        <div className={styles.header}>
          {title && (
            <h2 className={styles.title} id={titleId}>
              {title}
            </h2>
          )}
        </div>
        <div className={styles.content}>{children}</div>
      </div>
    </div>
  );
}
