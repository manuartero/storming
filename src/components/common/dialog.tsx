import c from "classnames";
import { DialogContent, DialogOverlay } from "@reach/dialog";
import type { HTMLAttributes, ReactElement } from "react";

import "@reach/dialog/styles.css";
import "./dialog.scss";

interface Props extends HTMLAttributes<HTMLElement> {
  children: ReactElement[];
  isOpen?: boolean;
  size?: "regular" | "large";
  onClose?: () => void;
}

export function Dialog(props: Props): JSX.Element {
  return (
    <DialogOverlay
      {...props}
      isOpen={props.isOpen}
      onDismiss={props.onClose}
      style={{ background: "hsla(0, 100%, 100%, 0.9)" }}
    >
      <DialogContent
        className={c(
          "dialog",
          props.size === "large" ? "dialog--large" : "dialog--regular"
        )}
      >
        {props.children}
      </DialogContent>
    </DialogOverlay>
  );
}
