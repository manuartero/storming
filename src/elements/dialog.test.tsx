import { render, screen } from "@testing-library/react";
import { Dialog } from "./dialog";

describe("<Dialog />", () => {
  test("render: dialog role [no title]", () => {
    render(<Dialog onClose={jest.fn()} />);

    const dialog = screen.getByRole("dialog");
    expect(dialog).toMatchSnapshot();
  });

  test("render: dialog role [with title]", () => {
    render(<Dialog title="Test Dialog" onClose={jest.fn()} />);

    const dialog = screen.getByRole("dialog");
    expect(dialog).toMatchSnapshot();
  });

  test("calls onClose() when exit button is clicked", () => {
    const onClose = jest.fn();
    render(<Dialog title="Test Dialog" onClose={onClose} />);

    const exitButton = screen.getByLabelText("close dialog");
    exitButton.click();

    expect(onClose).toHaveBeenCalledTimes(1);
  });
});
