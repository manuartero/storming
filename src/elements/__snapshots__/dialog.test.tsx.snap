// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`<Dialog /> render: dialog role [no title] 1`] = `
<div
  aria-modal="true"
  class="dialogScreen"
  role="dialog"
  tabindex="-1"
>
  <div
    class="dialog regular"
  >
    <button
      aria-label="close dialog"
      class="exitButton"
    >
      X
    </button>
    <div
      class="header"
    />
    <div
      class="content"
    />
  </div>
</div>
`;

exports[`<Dialog /> render: dialog role [with title] 1`] = `
<div
  aria-labelledby="dialog-title-«r0»"
  aria-modal="true"
  class="dialogScreen"
  role="dialog"
  tabindex="-1"
>
  <div
    class="dialog regular"
  >
    <button
      aria-label="close dialog"
      class="exitButton"
    >
      X
    </button>
    <div
      class="header"
    >
      <h2
        class="title"
        id="dialog-title-«r0»"
      >
        Test Dialog
      </h2>
    </div>
    <div
      class="content"
    />
  </div>
</div>
`;
